ParabolInc/parabol
# chore(metrics): Add client side metric Amplitude tracking

# Description

Implements #8678 to also send client side metrics to Amplitude using the Amplitude client library. Note that this PR does not remove the old client -> server -> Segment route just yet. We follow the migration pattern of parallel writing then deletion of the old service.

The real meat of the change is in `packages/client/mutations/SendClientSideEvent.ts`, which is renamed from `packages/client/mutations/SendClientSegmentEventMutation.ts`

## Demo

`Your Work Drawer Impression` is a client-side metric and notice the library used is the Amplitude library, instead of Segment.

<img width="1358" alt="Screenshot 2023-10-19 at 16 25 20" src="https://github.com/ParabolInc/parabol/assets/1879975/fdd5927d-70de-4297-96c2-f2be55ee4949">



## Testing scenarios

- [ ] Verify client side metric is emitted to Amplitude
  1.  Make sure you have the `AMPLITUDE_WRITE_KEY` in your `.env` (see [here](https://github.com/ParabolInc/action-devops/blob/a56919818da46a690cbfec98bf5cf09a6d61992f/environments/local#L1))
  2.  Start the server
  3.  Emit any client side metric (e.g., go to a stand up meeting page and play around with the Your Work Drawer)
  4.  Go to [Amplitude User Activity page](https://app.amplitude.com/analytics/parabol/activity)
  5.  Make sure you select Parabol Development on top left
  6.  Search your `userId`
  7.  On the User Look-Up Page, verify a couple of client side metrics are in the Event Stream

## Final checklist
